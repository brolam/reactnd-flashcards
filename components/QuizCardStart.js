import React from 'react'
import { StyleSheet } from 'react-native'
import { View, Text, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'
import cardStyle from '../styles/cardStyles'

export default function QuizCardStart({ deck, quizzes, onStart = () => { } }) {
  return (
    <View style={cardStyle}>
      <Text style={styles.temp}>{deck.title}</Text>
      <Text style={styles.temp}>{deck.amountOfCards} card(s)</Text>
      <TouchableOpacity  id="buttonStart" onPress={onStart}>
        <Text style={styles.temp} >Start</Text>
      </TouchableOpacity>
    </View>
  )
}

QuizCardStart.propTypes = {
  deck: PropTypes.shape({
    title: PropTypes.string.isRequired
  }),
  quizzes: PropTypes.array.isRequired,
}

const styles = StyleSheet.create({
  temp: { fontSize: 24, textAlign: 'center', paddingTop: 8 },
});