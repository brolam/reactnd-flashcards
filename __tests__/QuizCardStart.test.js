import React from 'react'
import QuizCardStart from '../components/QuizCardStart'

const quizzesDummy = [
  { question: 'One Quetion' },
  { question: 'Two Quetion' },
  { question: 'Three Quetion' }
]
const deckDummy = { id: 'one-item', title: 'One Deck', amountOfCards: 10, quizzes: quizzesDummy }

test('renders without crashing', () => {
  const _24HoursInMilliseconds = 86400000
  const deckDummyScore99Percent = {
    ...quizzesDummy,
    score: 99.99,
    lastUpdated: Date.now() - _24HoursInMilliseconds
  }
  const wrapper = shallow(<QuizCardStart deck={deckDummyScore99Percent} quizzes={quizzesDummy} />);
  expect(wrapper).toMatchSnapshot();
});

test('onStart event', () => {
  let onStart = jest.fn()
  const quizCardStart = shallow(
    <QuizCardStart
      deck={deckDummy}
      quizzes={quizzesDummy}
      onStart={onStart}
    />);
  quizCardStart.find('TouchableOpacity [id="buttonStart"]').simulate('press')
  expect(onStart).toHaveBeenCalled();
});

test('renders today score', () => {
  const deckDummyScore99Percent = {
    ...quizzesDummy,
    score: 99,
    lastUpdated: Date.now()
  }
  const quizCardStart = shallow(
    <QuizCardStart
      deck={deckDummyScore99Percent}
      quizzes={quizzesDummy}
    />);
  expect(quizCardStart).toMatchSnapshot();
});
