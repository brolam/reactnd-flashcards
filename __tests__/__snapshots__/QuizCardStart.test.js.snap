// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders today score 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 8,
          "padding": 20,
          "shadowColor": "rgba(0, 0, 0, 0.24)",
          "shadowOpacity": 0.8,
        },
        Object {
          "borderRadius": 8,
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowRadius": 3,
        },
      ],
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginBottom": 10,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 32,
        "fontWeight": "700",
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Today's Score 99.00%
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 42,
        "fontWeight": "700",
        "marginBottom": 5,
        "padding": 20,
        "textAlign": "center",
      }
    }
  >
     cards
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "space-around",
        "marginBottom": 5,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      id="buttonStart"
      onPress={[Function]}
      style={
        Array [
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#292477",
              "borderRadius": 3,
              "borderWidth": 1,
              "padding": 5,
              "paddingLeft": 25,
              "paddingRight": 25,
            },
          ],
          Object {
            "backgroundColor": "#3cb371",
            "flex": 1,
            "margin": 5,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 28,
            "fontWeight": "600",
          }
        }
      >
        Start
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      id="buttonAddQuiz"
      onPress={[Function]}
      style={
        Array [
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#292477",
              "borderRadius": 3,
              "borderWidth": 1,
              "padding": 5,
              "paddingLeft": 25,
              "paddingRight": 25,
            },
          ],
          Object {
            "backgroundColor": "#f26f28",
            "flex": 1,
            "margin": 5,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 28,
            "fontWeight": "600",
          }
        }
      >
        Add
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`renders without crashing 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 8,
          "padding": 20,
          "shadowColor": "rgba(0, 0, 0, 0.24)",
          "shadowOpacity": 0.8,
        },
        Object {
          "borderRadius": 8,
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowRadius": 3,
        },
      ],
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginBottom": 10,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 32,
        "fontWeight": "700",
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Last Score 99.99%
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 42,
        "fontWeight": "700",
        "marginBottom": 5,
        "padding": 20,
        "textAlign": "center",
      }
    }
  >
     cards
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "space-around",
        "marginBottom": 5,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      id="buttonStart"
      onPress={[Function]}
      style={
        Array [
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#292477",
              "borderRadius": 3,
              "borderWidth": 1,
              "padding": 5,
              "paddingLeft": 25,
              "paddingRight": 25,
            },
          ],
          Object {
            "backgroundColor": "#3cb371",
            "flex": 1,
            "margin": 5,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 28,
            "fontWeight": "600",
          }
        }
      >
        Start
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      id="buttonAddQuiz"
      onPress={[Function]}
      style={
        Array [
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#292477",
              "borderRadius": 3,
              "borderWidth": 1,
              "padding": 5,
              "paddingLeft": 25,
              "paddingRight": 25,
            },
          ],
          Object {
            "backgroundColor": "#f26f28",
            "flex": 1,
            "margin": 5,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 28,
            "fontWeight": "600",
          }
        }
      >
        Add
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;
